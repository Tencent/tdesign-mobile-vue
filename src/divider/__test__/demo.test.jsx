/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import baseVue from '@/divider/demos/base.vue';
import mobileVue from '@/divider/demos/mobile.vue';
import themeVue from '@/divider/demos/theme.vue';

const mapper = {
  baseVue,
  mobileVue,
  themeVue,
};

describe('Divider', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Divider ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
