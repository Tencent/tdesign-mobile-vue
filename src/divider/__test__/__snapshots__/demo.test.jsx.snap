// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Divider > Divider baseVue demo works fine 1`] = `
<div
  data-v-589c0842=""
>
  <div
    class="divider-demo__title"
    data-v-589c0842=""
  >
    水平分割线
  </div>
  <div
    class="t-divider t-divider--horizontal t-divider--center"
    data-v-589c0842=""
    role="separator"
  >
    <div
      class="t-divider__content"
    >
      <!---->
    </div>
  </div>
  <div
    class="divider-demo__title"
    data-v-589c0842=""
  >
    带文字水平分割线
  </div>
  <div
    class="t-divider t-divider--horizontal t-divider--left"
    data-v-589c0842=""
    role="separator"
  >
    <div
      class="t-divider__content"
    >
      文字信息
    </div>
  </div>
  <div
    class="t-divider t-divider--horizontal t-divider--center"
    data-v-589c0842=""
    role="separator"
  >
    <div
      class="t-divider__content"
    >
      文字信息
    </div>
  </div>
  <div
    class="t-divider t-divider--horizontal t-divider--right"
    data-v-589c0842=""
    role="separator"
  >
    <div
      class="t-divider__content"
    >
      文字信息
    </div>
  </div>
  <div
    class="divider-demo__title"
    data-v-589c0842=""
    style="margin-bottom: 10px;"
  >
    垂直分割线
  </div>
  <view
    class="divider-wrapper"
    data-v-589c0842=""
  >
    <span
      data-v-589c0842=""
    >
      文字信息
    </span>
    <div
      class="t-divider t-divider--vertical t-divider--center"
      data-v-589c0842=""
      role="separator"
    >
      <div
        class="t-divider__content"
      >
        <!---->
      </div>
    </div>
    <span
      data-v-589c0842=""
    >
      文字信息
    </span>
    <div
      class="t-divider t-divider--vertical t-divider--center"
      data-v-589c0842=""
      role="separator"
    >
      <div
        class="t-divider__content"
      >
        <!---->
      </div>
    </div>
    <span
      data-v-589c0842=""
    >
      文字信息
    </span>
  </view>
</div>
`;

exports[`Divider > Divider mobileVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
  data-v-bf2b60a0=""
>
  <h1
    class="title"
    data-v-bf2b60a0=""
  >
    Divider 分割符
  </h1>
  <p
    class="summary"
    data-v-bf2b60a0=""
  >
    用于分割、组织、细化有一定逻辑的组织元素内容和页面结构。
  </p>
  <div
    class="tdesign-mobile-demo-block"
    data-v-bf2b60a0=""
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        01 组件类型
      </h2>
      <!--v-if-->
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        data-v-589c0842=""
        data-v-bf2b60a0=""
      >
        <div
          class="divider-demo__title"
          data-v-589c0842=""
        >
          水平分割线
        </div>
        <div
          class="t-divider t-divider--horizontal t-divider--center"
          data-v-589c0842=""
          role="separator"
        >
          <div
            class="t-divider__content"
          >
            <!---->
          </div>
        </div>
        <div
          class="divider-demo__title"
          data-v-589c0842=""
        >
          带文字水平分割线
        </div>
        <div
          class="t-divider t-divider--horizontal t-divider--left"
          data-v-589c0842=""
          role="separator"
        >
          <div
            class="t-divider__content"
          >
            文字信息
          </div>
        </div>
        <div
          class="t-divider t-divider--horizontal t-divider--center"
          data-v-589c0842=""
          role="separator"
        >
          <div
            class="t-divider__content"
          >
            文字信息
          </div>
        </div>
        <div
          class="t-divider t-divider--horizontal t-divider--right"
          data-v-589c0842=""
          role="separator"
        >
          <div
            class="t-divider__content"
          >
            文字信息
          </div>
        </div>
        <div
          class="divider-demo__title"
          data-v-589c0842=""
          style="margin-bottom: 10px;"
        >
          垂直分割线
        </div>
        <view
          class="divider-wrapper"
          data-v-589c0842=""
        >
          <span
            data-v-589c0842=""
          >
            文字信息
          </span>
          <div
            class="t-divider t-divider--vertical t-divider--center"
            data-v-589c0842=""
            role="separator"
          >
            <div
              class="t-divider__content"
            >
              <!---->
            </div>
          </div>
          <span
            data-v-589c0842=""
          >
            文字信息
          </span>
          <div
            class="t-divider t-divider--vertical t-divider--center"
            data-v-589c0842=""
            role="separator"
          >
            <div
              class="t-divider__content"
            >
              <!---->
            </div>
          </div>
          <span
            data-v-589c0842=""
          >
            文字信息
          </span>
        </view>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block"
    data-v-bf2b60a0=""
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        02 组件状态
      </h2>
      <!--v-if-->
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        data-v-127b67ab=""
        data-v-bf2b60a0=""
      >
        <div
          class="divider-demo__title"
          data-v-127b67ab=""
        >
          虚线样式
        </div>
        <div
          class="t-divider t-divider--horizontal t-divider--center t-divider--dashed"
          data-v-127b67ab=""
          role="separator"
        >
          <div
            class="t-divider__content"
          >
            <!---->
          </div>
        </div>
        <div
          class="t-divider t-divider--horizontal t-divider--left t-divider--dashed"
          data-v-127b67ab=""
          role="separator"
        >
          <div
            class="t-divider__content"
          >
            文字信息
          </div>
        </div>
        <div
          class="t-divider t-divider--horizontal t-divider--center t-divider--dashed"
          data-v-127b67ab=""
          role="separator"
        >
          <div
            class="t-divider__content"
          >
            文字信息
          </div>
        </div>
        <div
          class="t-divider t-divider--horizontal t-divider--right t-divider--dashed"
          data-v-127b67ab=""
          role="separator"
        >
          <div
            class="t-divider__content"
          >
            文字信息
          </div>
        </div>
      </div>
      
    </div>
  </div>
</div>
`;

exports[`Divider > Divider themeVue demo works fine 1`] = `
<div
  data-v-127b67ab=""
>
  <div
    class="divider-demo__title"
    data-v-127b67ab=""
  >
    虚线样式
  </div>
  <div
    class="t-divider t-divider--horizontal t-divider--center t-divider--dashed"
    data-v-127b67ab=""
    role="separator"
  >
    <div
      class="t-divider__content"
    >
      <!---->
    </div>
  </div>
  <div
    class="t-divider t-divider--horizontal t-divider--left t-divider--dashed"
    data-v-127b67ab=""
    role="separator"
  >
    <div
      class="t-divider__content"
    >
      文字信息
    </div>
  </div>
  <div
    class="t-divider t-divider--horizontal t-divider--center t-divider--dashed"
    data-v-127b67ab=""
    role="separator"
  >
    <div
      class="t-divider__content"
    >
      文字信息
    </div>
  </div>
  <div
    class="t-divider t-divider--horizontal t-divider--right t-divider--dashed"
    data-v-127b67ab=""
    role="separator"
  >
    <div
      class="t-divider__content"
    >
      文字信息
    </div>
  </div>
</div>
`;
